bug: Missing Duration import breaks coverage tests

## Summary

The Coverage workflow is failing on main due to a missing `Duration` import in `shard_group_tracker.rs`.

## Error

```
error[E0412]: cannot find type `Duration` in this scope
   --> crates/gglib-download/src/manager/shard_group_tracker.rs:188:39
    |
188 |     pub fn gc_expired(&mut self, ttl: Duration) -> usize {
    |                                       ^^^^^^^^ not found in this scope
```

## Root Cause

PR #110 (commit f58407c) moved `gc_expired()` to be test-only with `#[cfg(test)]` and moved the `Duration` import into the test module. However, `gc_expired()` is defined in the main impl block (outside the test module), so it cannot access the `Duration` type imported only within `mod tests`.

The issue was introduced in commit 4dacc54 within PR #110, which stated:
> fix: remove unused Duration import
> Move Duration import to test module since gc_expired is now test-only

The import was moved to the wrong location - it should have been kept at the module level since `#[cfg(test)]` methods in the main impl block still need access to module-level imports.

## Impact

- ❌ Coverage workflow failing on main
- ❌ Blocks merging of any new PRs requiring coverage checks
- ✅ Regular CI tests pass (they don't compile with coverage instrumentation)

## References

- Introduced in: PR #110, commit 4dacc54
- Failed workflow: https://github.com/mmogr/gglib/actions/runs/21333580788

--- a/crates/gglib-download/src/manager/shard_group_tracker.rs
+++ b/crates/gglib-download/src/manager/shard_group_tracker.rs
@@ -7,6 +7,9 @@ use std::collections::HashMap;
 use std::path::PathBuf;
 use std::time::Instant;
 
+#[cfg(test)]
+use std::time::Duration;
+
 use gglib_core::download::Quantization;
 
 use crate::queue::ShardGroupId;
@@ -199,7 +202,6 @@ impl ShardGroupTracker {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use std::time::Duration;
 
     fn test_metadata() -> GroupMetadata {
         GroupMetadata {

